Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -104,205 +206,432 @@ if (-not($exchangeShell.ShellLoaded)) {
+                                $cert = $null
+                                $cert = $signature.SignerCertificate
+                                if ( $null -eq $cert) {
+                                    $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                    Write-SimpleLogFile -String ("[FAIL] - We could not get signer certificate on Module: $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                                } else {
+                                    $chain = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Chain
+                                    $chain.Build($cert) | Out-Null
+                                    $rootCertificate = $null
+                                    $rootCertificate = $chain.ChainElements[$chain.ChainElements.Count - 1].Certificate.Subject
+                                    if ( $null -eq $rootCertificate) {
+                                        $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                        Write-SimpleLogFile -String ("[FAIL] - We could not get root certificate on Module: $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                                    } else {
+                                        $FIPSPath = Join-Path $env:ExchangeInstallPath 'FIP-FS\Bin\TE'
+                                        if ($module.FileName.ToLower().StartsWith($FIPSPath.ToLower()) ) {
+                                            if ($Offline) {
+                                                if ( -not (CheckIfISAcceptedRootCA -CAString $rootCertificate -isFIPFS -Offline) ) {
+                                                    $BadProcessList["$($Instance.Id) - $($module.FileName)"] = "$($Instance.Id), $($module.FileName), $($module.Company), $($module.Description), $($module.Product),  $($Instance.Path), $CommandLine"
+                                                    Write-SimpleLogFile -String ("[FAIL] - root do not expected (FIP-FS) on Module: $($module.FileName) `n`tCompany: $($module.Company) `n`tDescription: $($module.Description) `n`tProduct: $($module.Product) `n`tRoot CA $rootCertificate `n`tProcess ID: $($Instance.Id) `n`tCommand Line: $CommandLine") -name $LogFile -OutHost
+                                                }
```

Comment:
Why are we checking to see if the module is signed correctly in this script? 

This script is to check for AV Exclusions the easiest way possible. If there is a threat concern that we are trying to validate here. That would need to be done in a different script, not in an AV Exclusions script.

Reply:
We are checking that the modules are signed correctly by well-know CAs.
```
CN=Microsoft Corporate Root CA, O=Microsoft Corporation
CN=Microsoft Root Authority, OU=Microsoft Corporation, OU=Copyright (c) 1997 Microsoft Corp.
CN=Microsoft Root Certificate Authority, DC=microsoft, DC=com
CN=Microsoft Root Certificate Authority 2010, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
CN=Microsoft Root Certificate Authority 2011, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
```
In case or FIPS files Verisign is valid as well.
```
CN=VeriSign Class 3 Public Primary Certification Authority - G5, OU="(c) 2006 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
```
I think it is better to use the signature than the company name in the file.
In this way, we can verify that the module is not from a different company and that our modules were not modified.